File: semgrep-core/tests/patterns/java/dots_method_chaining.java
class Foo {

  void test() {

    //ERROR: match
    f = this.foo().m().h().bar().z();

    //ERROR: match
    f = this.foo().bar();

    f = this.foo().m().h().z();

    //ERROR: match $O can match o.before()
    f = this.before().foo().m().h().bar().z();
  }
}

File: semgrep-core/tests/patterns/rust/dots_method_chaining.rs
use std::process::Command;

fn main() -> io::Result<()> {
    // ERROR: match
    let proc = Command::new("semgrep-core")
        .args(["-l", "rust"])
        .args(["-rules", "test.yaml"])
        .output()
        .expect("failed to execute process");

    Ok(())
}
